### PR TITLE
[iOS] Record add code example

### PR DESCRIPTION
**Pull request by HenryHins**
1. added code example
connects #173 

2. Added Reserved Column
connects #160 

3. Fixed links in Records
connects #141 

4. Add Data Type
connects #214 

5. Make SKYRecord explanation more consistent with JS Guide
connects #215

### CODE REVIEW

File: content/ios/record.md [hunk 1]
```diff
@@ -182,11 +237,28 @@ It will establish a reference from _Record B_ to _Record A_.
 <a name="data-type"></a>
 ## Data Type
 
-<!--- TODO: link to server here -->
+Skygear supports almost all of the builtin JavaScript types, such as:
```
Comment (ben181231) [inline content/ios/record.md]: @HenryHins JavaScript?

File: content/ios/record.md [hunk 2]
```diff
@@ -2,23 +2,28 @@
...
 - `SKYRecord` must have a type.
-- `SKYRecord` is a key-value data object that can be stored at _database_.
-- `SKYRecord` will belong to the currently logged in user.
+- Each `SKYRecord ` object is like a dictionary with keys and values; keys will be mapped to database column names, and values will be stored appropriately
+based on the data type. Please refer to [Data Type](#data-type)
+section within this guide for more information.
+- `SKYRecord` will be owned by the currently logged in user.
 - `SKYRecord` object has a unique `id` (a string combination of record type and uuid is used).
 - Each `SKYRecord` has a `recordType`, which describes the _type_ of data this record holds.
+-`Record` has reserved keys that cannot be used, such as `ownerUserRecordID ` and `recordType `. Please refer to [Reserved Columns](#reserved-columns) section for more.
```
Comment (ben181231) [inline content/ios/record.md]: The format seems not correct. This line will be appended to the previous line.

Comment (ben181231) [inline content/ios/record.md]: By the way, it should be `SKYRecord`, not `Record`, right?

Comment (HenryHins) [inline content/ios/record.md]: Yeah it should be. Sorry for mixing up the iOS guide with the JS one.

**Comment by ben181231** [comment]
merged as c2b92c4

**Pull request closed without merging**